cpp/src/join/hash_join.cu

```diff
@@ -151,40 +107,50 @@ get_left_join_indices_complement(rmm::device_vector<size_type> &right_indices,
+                               invalid_index_map->end(),
+                               int32_t{1});
+
     // Functor to check for index validity since left joins can create invalid indices
     valid_range<size_type> valid(0, right_table_row_count);
 
     // invalid_index_map[index_ptr[i]] = 0 for i = 0 to right_table_row_count
     // Thus specifying that those locations are valid
     thrust::scatter_if(rmm::exec_policy(stream),
                        thrust::make_constant_iterator(0),
-                       thrust::make_constant_iterator(0) + right_indices.size(),
-                       right_indices.begin(),      // Index locations
-                       right_indices.begin(),      // Stencil - Check if index location is valid
-                       invalid_index_map.begin(),  // Output indices
-                       valid);                     // Stencil Predicate
+                       thrust::make_constant_iterator(0) + right_indices->size(),
+                       right_indices->begin(),      // Index locations
+                       right_indices->begin(),      // Stencil - Check if index location is valid
+                       invalid_index_map->begin(),  // Output indices
+                       valid);                      // Stencil Predicate
     size_type begin_counter = static_cast<size_type>(0);
     size_type end_counter   = static_cast<size_type>(right_table_row_count);
 
     // Create list of indices that have been marked as invalid
     size_type indices_count = thrust::copy_if(rmm::exec_policy(stream),
                                               thrust::make_counting_iterator(begin_counter),
                                               thrust::make_counting_iterator(end_counter),
-                                              invalid_index_map.begin(),
-                                              right_indices_complement.begin(),
+                                              invalid_index_map->begin(),
+                                              right_indices_complement->begin(),
                                               thrust::identity<size_type>()) -
-                              right_indices_complement.begin();
-    right_indices_complement.resize(indices_count);
+                              right_indices_complement->begin();
+    right_indices_complement->resize(indices_count, stream);
   }
 
-  rmm::device_vector<size_type> left_invalid_indices(right_indices_complement.size(),
-                                                     JoinNoneValue);
+  auto left_invalid_indices =
+    std::make_unique<rmm::device_uvector<size_type>>(right_indices_complement->size(), stream);
+  thrust::fill(rmm::exec_policy(stream),
+               left_invalid_indices->begin(),
+               left_invalid_indices->end(),
+               JoinNoneValue);
 
   return std::make_pair(std::move(left_invalid_indices), std::move(right_indices_complement));
```

Comment:
I think you can remove the `std::move` here too? @jrhemstad ?

Reply:
The `pair` is taking ownership of the objects, so the move is needed her.

Reply:
So the compiler can't elide the copies when you copy into a pair, return the pair, and delete the stack-allocated things you copied?